blockchain/blocks/src/header.rs

```diff
@@ -339,6 +340,48 @@ impl BlockHeader {
         // h(vrfout) * totalPower < e * sectorSize * 2^256
         lhs < rhs
     }
+
+    pub async fn validate_block_drand<B: Beacon>(
+        &self,
+        beacon: Arc<B>,
+        prev_entry: BeaconEntry,
+    ) -> Result<(), Error> {
+        let max_round = beacon.max_beacon_round_for_epoch(self.epoch, &prev_entry);
+        if max_round == prev_entry.round() {
+            if !self.beacon_entries.is_empty() {
+                return Err(Error::Validation(format!(
+                    "expected not to have any beacon entries in this block, got: {:?}",
+                    self.beacon_entries.len()
+                )));
+            }
+            return Ok(());
+        }
+
+        let last = self.beacon_entries.last().unwrap();
+        if last.round() != max_round {
+            return Err(Error::Validation(format!(
+                "expected final beacon entry in block to be at round {}, got: {}",
+                max_round,
+                last.round()
+            )));
+        }
+        self.beacon_entries.iter().try_fold(
+            (true, &prev_entry),
+            |prev, curr| -> Result<(bool, &BeaconEntry), Error> {
+                let valid = beacon
+                    .verify_entry(curr, &prev.1)
+                    .map_err(|e| Error::Validation(e.to_string()))?;
+                if !valid {
+                    return Err(Error::Validation(format!(
+                        "beacon entry was invalid: curr:{:?}, prev: {:?}",
+                        curr, prev.1
+                    )));
+                }
+                Ok((valid, curr))
+            },
+        )?;
```

Comment:
It seems like you're not using the boolean flag here anywhere (and I think it's always `true` anyway), so you should be able to just get rid of it

Reply:
Oh yeah, my brain got folded up trying to reason with `fold` haha